product-service/lambda/createProduct.ts

```diff
@@ -0,0 +1,109 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { randomUUID } from "crypto";
+
+const client = new DynamoDBClient({});
+const ddbDocClient = DynamoDBDocumentClient.from(client);
+
+const PRODUCTS_TABLE_NAME = process.env.PRODUCTS_TABLE_NAME!;
+const STOCKS_TABLE_NAME = process.env.STOCKS_TABLE_NAME!;
+
+interface Product {
+  id: string;
+  title: string;
+  description: string;
+  price: number;
+}
+
+interface Stock {
+  product_id: string;
+  count: number;
+}
+
+export const handler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  console.log("Received event:", JSON.stringify(event, null, 2));
+
+  const headers = {
+    "Content-Type": "application/json",
+    "Access-Control-Allow-Origin": "*",
+    "Access-Control-Allow-Methods": "POST",
+  };
+
+  try {
+    const body = JSON.parse(event.body || "{}");
+    const { title, description, price, count } = body;
+
+    if (
+      !title ||
+      !description ||
+      typeof price !== "number" ||
+      typeof count !== "number" ||
+      count < 0 ||
+      price <= 0
+    ) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ error: "Invalid product data" }),
+        headers,
+      };
+    }
```

Comment:
this validation can be further improved, currently I can still save some inappropriate data


<img width="558" alt="Screenshot 2024-06-24 at 15 02 35" src="https://github.com/hasanovanar/backend-nodejs-aws-shop-react/assets/64736269/0dbea500-1e5e-4bc1-8c53-ced90c7bbf9b">